specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
 
 == Use case examples
 
-=== Basic non-virtualized system
+This chapter provides guidance for building secure Risc-V systems for a selection of common use cases. It is not intended to be an exhaustive selection. The purpose is to illustrate common principles that can be applied in a wide variety of use cases. 
+
+Additional use cases may be added in future versions.
+
+=== Generic system
 
 ==== Overview
 
-==== Secure and Verified Boot
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Generic vertically integrated system"]
+image::img_ch4_priv.png[]
+
+A generic vertically integrated system can be either virtualized or non-virtualized. 
+
+M-mode hosts a FW RoT. An OS or a Hypervisor in S or HS mode controls applications or guests. Guests and applications execute in U or VS/VU modes and trust the OS or Hypervisor to provide isolation guarantees.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN 
+| A secure system SHOULD support either S mode, or the hypervisor extension (HS/VS/VU)
+
+|===
+
+NOTE: The RISC-V architecture also caters for systems with just M and U modes, commonly used in embedded systems, helper cores, and similar use cases. On secure systems not supporting S mode a FW RoT has to share M-mode with an OS. RISC-V does not exclude such implementations - for example, implementations using a certified OS and FW RoT, or using a HW RoT to isolate sensitive code and assets (physical isolation). There is no current mechanism in RISC-V for isolation within M-mode itself other than temporal boundaries +
+ +
+To minimize the TCB of the FW RoT, RISC-V recommends that secure systems implement S mode and de-privilege the OS. 
 
 ==== Isolation model
 
-==== Device access control
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
 
-==== Sealing
+| CAT_NNN 
+| PMP/ePMP MUST be used to isolate M-mode from lower privilege levels.
+
+| CAT_NNN 
+| sPMP or MMU MUST be used to isolate user applications or virtual guests
+
+|===
+
+Using sPMP is typically a more static model and can achieve a more deterministic system, for example in automotive or automation use cases. 
+
+MMU is typically required for Linux based systems, for example MMI use cases or edge devices.
+
+Either MMU and sPMP can be used both with or without hypervisor extension. For example, the hypervisor extension with sPMP can support static partition hypervisors, commonly used in automotive. And a single stage MMU can be used without hypervisor extension for full Linux support.
+
+==== Root of Trust
+
+See xref:chapter2.adoc#_reference_model[reference model].
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN 
+| A secure system SHOULD implement a HW RoT
+
+|===
+
+==== Authorized Boot
+
+Multiple models can be used to ensure a secure system can only run authorized software.
```

Comment:
authenticated software?

Reply:
"Authorized" software is the generic term used throughout the document. It is defined in chapter 2 (ecosystem model), with all the combinations and permutations of measured and verified, local or remote authorization, etc.

I appreciate different ecosystems have slightly different taxonomy. Basically it is impossible to not offend someone. Chapter 2 should allow people to map to whatever taxonomy they happen to be used to, in whatever ecosystem they happen to come from.